### PR TITLE
Add custom error type for routing error

### PR DESCRIPTION
**Pull request by jackholmes1010**
Throw custom error if route is not found.

This will allow you to handled 404's like this:

```typescript
const middleware: MiddlewareFunction = async (event, context, next) => {
    try {
        return await next(event, context)
    } catch (err) {
        if (err instanceof RouterError && err.code === "ROUTE_NOT_FOUND") {
            return {
                statusCode: 404,
                body: JSON.stringify({ code: err.code }),
            }
        }

        throw err
    }
}
```

### CODE REVIEW

File: src/RouterError.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { RouterErrorCode } from "./types"
+
+/**
+ * Represents a failure to route an event.
+ */
+export default class RouterError extends Error {
+    public isRequestParseError = true
```
Comment (johnyherangi) [inline src/RouterError.ts]: Is this required?

Comment (jackholmes1010) [reply src/RouterError.ts]: yee my bad

Comment (jackholmes1010) [reply src/RouterError.ts]: i changed it to `isRouterError`. The reason being is then you can also check `error.isRouterError`, as well as `err instanceof RouterError`.

**Review by WilsonZheng — APPROVED** [review]

**Review by johnmikk — APPROVED** [review]
Looks good

**Review by johnyherangi — APPROVED** [review]

**Pull request merged**